ui/components/multichain/account-list-item-menu/account-list-item-menu.js

```diff
@@ -178,19 +178,29 @@ export const AccountListItemMenu = ({
     >
       <ModalFocus restoreFocus initialFocusRef={anchorElement}>
         <div onKeyDown={handleKeyDown} ref={popoverDialogRef}>
-          {shouldRenderConnectAccount ? (
+          {process.env.MULTICHAIN ? (
             <Box display={[Display.Flex, Display.None]}>
               <MenuItem
                 data-testid="account-list-menu-connect-account"
                 onClick={() => {
-                  dispatch(
-                    addPermittedAccount(activeTabOrigin, identity.address),
-                  );
+                  if (isConnected) {
+                    dispatch(
+                      removePermittedAccount(activeTabOrigin, identity.address),
+                    );
+                  } else {
+                    dispatch(
+                      addPermittedAccount(activeTabOrigin, identity.address),
```

Comment:
Should we also switch to this account?

Reply:
I dont think it should. Otherwise we might confuse the user with the meaning of `Connecting` and `Switching` an account

Reply:
Good shout!